File: frontend/devpulse-ui/src/app/core/models/manager-dashboard-dto.ts
import { LabelNumberDto, SummaryCardsDto } from './developer-dashboard-dto';

export interface ManagerDashboardDto {
  // summary - Cards
  summaryCardsDto: ManagerSummaryCardsDto;

  // Team Journals Per Developer (Bar)
  teamJournalsPerDeveloperBarChartDto: LabelNumberDto[];

  // Feedback Given vs Pending (Donut)
  feedbackDonutChartDto: FeedbackDonutChartDto;

  // Tasks Assigned vs Completed (Stacked Bar)
  tasksWithStatus: LabelNumberDto[];

  // last updated time
  lastUpdated: string;

  // other information
  managerId: string;
  managerDisplayName: string;
}

export interface FeedbackDonutChartDto {
  feedbackCompleted: number;
  feedbackPending: number;
}

export interface ManagerSummaryCardsDto {
  highPriorityCount: number;
  newTasksCount: number;
  inProgressTasksCount: number;
  urgentTasksCount: number;
  newJournalsNeedingFeedback: number;
}

File: backend/SharedLib/DTOs/ManagerDashboard/ManagerDashboardDto.cs
namespace SharedLib.DTOs.ManagerDashboard
{
    public record ManagerDashboardDto(
    ManagerSummaryCardsDto SummaryCardsDto,
    List<LabelNumberDto> TeamJournalsPerDeveloperBarChartDto,
    FeedbackDonutChartDto FeedbackDonutChartDto,
    List<LabelNumberDto> TasksWithStatus,
    string LastUpdated,
    Guid ManagerId,
    string ManagerDisplayName);

    public record ManagerSummaryCardsDto(
        int HighPriorityCount,
        int NewTasksCount,
        int InProgressTasksCount,
        int UrgentTasksCount,
        int NewJournalsNeedingFeedback
    );

    public record FeedbackDonutChartDto(
        int FeedbackCompleted,
        int FeedbackPending
    );

    public record LabelNumberDto(
        string Label,
        int Value
    );
}
